### ISSUE TITLE
Bot#emoji and multiple servers

### ISSUE THREAD

**Issue by Mijyuoon** [meew0/discordrb#397]
The `Bot#emoji` method doesn't work properly when multiple servers have emoji with same name. It properly registers only one of them due to [this code](https://github.com/meew0/discordrb/blob/3bd10f9d9fd868fe5a84838628fe8ca0e7c55ad7/lib/discordrb/bot.rb#L174) using a hashmap with emoji name as the key.

**Comment by z64** [comment]
Some eval that demonstrates this:
```
[1] pry(Powerbot)> BOT.emoji.select { |e| e.name == 'patty' }
=> [<GlobalEmoji name=patty id=230212690247811072>]
[3] pry(Powerbot)> BOT.server(225375815087554563).emoji.values.select { |e| e.name == 'patty' }
=> [<Emoji name=patty id=314086116812914689>]
[4] pry(Powerbot)> BOT.server(121851668936654848).emoji.values.select { |e| e.name == 'patty' }
=> [<Emoji name=patty id=230212690247811072>]
```
Moreover, `GlobalEmoji` should be removed - it is left over code from before custom emoji for everyone and not just twitch-partnered servers, etc, were a thing, and duplicates the same behavior of the other `Emoji` class.

**Comment by meew0** [comment]
Fixed in 7dfec1d
